crates/recursor/src/error.rs

```diff
@@ -216,45 +204,6 @@ impl From<Error> for String {
     }
 }
 
-impl From<ResolveError> for Error {
-    fn from(e: ResolveError) -> Self {
-        let nx_domain = e.is_nx_domain();
-        let no_records_found = e.is_no_records_found();
-
-        let proto_err = match ProtoErrorKind::try_from(e) {
-            Ok(res) => res,
-            Err(e) => return ErrorKind::Resolve(e).into(),
-        };
-
-        let ProtoErrorKind::NoRecordsFound(NoRecords {
-            query,
-            soa,
-            ns,
-            authorities,
-            ..
-        }) = proto_err
-        else {
-            return ErrorKind::Proto(proto_err.into()).into();
-        };
-
-        if let Some(ns) = ns {
-            ErrorKind::ForwardNS(ns).into()
```

Comment:
The tests are failing because penultimate iterative queries that get a referral response are now being turned into `NoRecordsFound` errors instead of `ForwardNS` errors, as expected here: https://github.com/hickory-dns/hickory-dns/blob/ab84df98a70416b98628624e61f5255140fd4ca3/crates/recursor/src/recursor_dns_handle.rs#L253. We'll need to re-introduce this error conversion logic or teach the recursor how to handle other error types.